src/Microsoft.DotNet.Build.CloudTestTasks/UploadClient.cs

```diff
@@ -208,6 +210,57 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                 }
             }
         }
+
+        public async Task<bool> FileEqualsExistingBlobAsync(
+            string accountName,
+            string accountKey,
+            string containerName,
+            string filePath,
+            string destinationBlob,
+            int uploadTimeout)
+        {
+            using (var client = new HttpClient
+            {
+                Timeout = TimeSpan.FromMinutes(uploadTimeout)
+            })
+            {
+                log.LogMessage(
+                    MessageImportance.Low,
+                    $"Downloading blob {destinationBlob} to check if identical.");
+
+                string blobUrl = AzureHelper.GetBlobRestUrl(accountName, containerName, destinationBlob);
+                var createRequest = AzureHelper.RequestMessage("GET", blobUrl, accountName, accountKey);
+
+                using (HttpResponseMessage response = await AzureHelper.RequestWithRetry(
+                    log,
+                    client,
+                    createRequest))
+                {
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        throw new HttpRequestException(
+                            $"Failed to retrieve existing blob {destinationBlob}, " +
+                            $"status code {response.StatusCode}.");
+                    }
+
+                    byte[] existingBytes = await response.Content.ReadAsByteArrayAsync();
+                    byte[] localBytes = File.ReadAllBytes(filePath);
+
+                    bool equal = localBytes.SequenceEqual(existingBytes);
```

Comment:
The only time this will ever even have a chance of being byte-for-byte equal is if they have the exact build numbers passed it. Is that going to be the case for your scenarios? If so OK but if it is for a rebuild scenario this comparison will not work. It will only likely work for a republish of the same packages scenario.

Reply:
Yeah, the only scenario I'm aiming at is package republish. Agreed this wouldn't work for anything else.